apis/compute/v1beta1/computetargettcpproxy_types.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	ComputeTargetTCPProxyGVK = schema.GroupVersionKind{
+		Group:   GroupVersion.Group,
+		Version: GroupVersion.Version,
+		Kind:    "ComputeTargetTCPProxy",
+	}
+)
+
+// ComputeTargetTCPProxySpec defines the desired state of ComputeTargetTCPProxy
+// +kcc:proto=google.cloud.compute.v1.TargetTcpProxy
+type ComputeTargetTCPProxySpec struct {
+	// A reference to the ComputeBackendService resource.
+	// +required
+	BackendServiceRef *refs.ComputeBackendServiceRef `json:"backendServiceRef"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Description is immutable"
+	// Immutable. An optional description of this resource.
+	Description *string `json:"description,omitempty"`
+
+	// The geographical location of the ComputeTargetTCPProxy.
+	// Reference: GCP definition of regions/zones (https://cloud.google.com/compute/docs/regions-zones/)
+	// +optional
+	Location *string `json:"location"`
```

Comment:
Have to explicitly add optional tag otherwise location will be marked as required field in the CRD, while it's not required for existing global tcp proxy, this change will break existing global tcp proxy resources.

I'm not familiar with how the CRD is generated and why location is marked as required by default. I noticed that in the previous generated types, we specify +optional but not +required([example](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/8dabfc844d7bb35d243216047f140f5d7e770441/pkg/clients/generated/apis/compute/v1beta1/computetargettcpproxy_types.go#L43)), is there any reason we switch to only specify +required not +optional now?